airflow/migrations/versions/bbf4a7ad0465_remove_id_column_from_xcom.py

```diff
@@ -43,9 +100,14 @@ def upgrade():
     with op.batch_alter_table('xcom') as bop:
         xcom_columns = [col.get('name') for col in inspector.get_columns("xcom")]
         if "id" in xcom_columns:
+            if conn.dialect.name == 'mssql':
+                constraint_dict = get_table_constraints(conn, "xcom")
+                drop_column_constraints(bop, 'id', constraint_dict)
             bop.drop_column('id')
             bop.drop_index('idx_xcom_dag_task_date')
-            bop.create_primary_key('pk_xcom', ['dag_id', 'task_id', 'key', 'execution_date'])
+            # mssql doesn't allow primary keys with nullable columns
+            if conn.dialect.name != 'mssql':
+                bop.create_primary_key('pk_xcom', ['dag_id', 'task_id', 'key', 'execution_date'])
```

Comment:
These columns _shouldn't_ be nullable, so lets fix that.

However since this migration has already been released, this change will have to stay, and you'll need to create a new migration that makes these cols non-null and adds the missing migration for MSSQL

Reply:
sure, will add another migration